Products/ZenUI3/browser/device_export.py

```diff
@@ -0,0 +1,99 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2015, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import logging
+from Products import Zuul
+from Products.Five.browser import BrowserView
+from Products.ZenUtils.jsonutils import unjson
+from Products.Zuul.facades.devicefacade import DeviceFacade
+from xml.etree.ElementTree import Element, tostring
+
+log = logging.getLogger('zen.deviceexporter')
+
+
+class DeviceExporter(BrowserView):
+
+    def __call__(self):
+        body = unjson(self.request.form['body'])
+        type = body.get('type')
+        getattr(self, type)(self.request.response, self._query(body), body['fields'])
+
+    def _query(self, params):
+        device_router = DeviceFacade(self.context)
+        devices = device_router.getDevices(sort=params['sort'], dir=params['sdir'], params=params['params'])
```

Comment:
This is only going to get the first 50 devices unless you specifically pass in a higher limit